### PR TITLE
Room restrictions

### PR DESCRIPTION
**Pull request by meluru**

### CODE REVIEW

File: indico/modules/rb/operations/admin.py [hunk 1]
```diff
@@ -56,8 +56,8 @@ def update_room_availability(room, availability):
 
 
 def update_room(room, args):
-    acl_entries = args.pop('acl_entries')
```
Comment (ThiefMaster) [inline indico/modules/rb/operations/admin.py]: you can just add a `, None` to the pop call to fix this

File: indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx [hunk 2]
```diff
@@ -766,6 +770,20 @@ class RoomEditModal extends React.Component {
                                 isEqual={_.isEqual}
                                 format={value => (value === null ? [] : value)} />
                 );
+            case 'roomVisibility':
+                return (
+                    <>
+                        <h5>{content.label}</h5>
+                        <Form.Group key={key} label={Translate.string('General')}>
+                            <FinalRadio name="roomVisibility"
```
Comment (ThiefMaster) [inline indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx]: what about using `protectionMode` here to match the name of the Room attribute?

File: indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx [hunk 3]
```diff
@@ -890,14 +894,17 @@ class RoomEditModal extends React.Component {
             reservationsNeedConfirmation: false,
             capacity: null,
             attributes: roomAttributes,
+            protectionMode: 'public',
             ...roomEquipment,
             ...roomAvailability,
         } : {
             ...room,
+            protectionMode: room.isPublic ? 'public' : 'protected',
```
Comment (ThiefMaster) [inline indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx]: just add the `protection_mode` in the marshmallow schema, then no extra handling is needed here

Comment (mic4ael) [reply indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx]: Yeah, I wasn't sure whether to use already exising `isPublic` or add `protection_mode` to the schema

File: indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx [hunk 4]
```diff
@@ -766,6 +770,23 @@ class RoomEditModal extends React.Component {
...
+                        <p className="field-description">
+                            Restricted rooms can only be booked by users defined in the room ACL
+                        </p>
+                        <Form.Group>
+                            <FinalRadio name="protectionMode"
+                                        value="public"
+                                        label={Translate.string('Public')} />
+                            <FinalRadio name="protectionMode"
+                                        value="protected"
+                                        label={Translate.string('Protected')} />
```
Comment (ThiefMaster) [inline indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx]: "Restricted"? That's more consistent with the description.

File: indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx [hunk 5]
```diff
@@ -766,6 +770,23 @@ class RoomEditModal extends React.Component {
                                 isEqual={_.isEqual}
                                 format={value => (value === null ? [] : value)} />
                 );
+            case 'protectionMode':
+                return (
+                    <React.Fragment key={key}>
+                        <h5>{content.label}</h5>
+                        <p className="field-description">
+                            Restricted rooms can only be booked by users defined in the room ACL
```
Comment (mvidalgarcia) [inline indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx]: i18n

Comment (mic4ael) [reply indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx]: Good catch!

**Pull request merged**